### ISSUE TITLE
Fatal Error

### ISSUE THREAD

**Issue by mimmi20** [endorphin-studio/browser-detector#17]
```php
PHP Fatal error:  Uncaught Symfony\Component\Yaml\Exception\ParseException: Numeric keys are not supported. Quote your evaluable mapping keys instead at line 10 (near "3.1: "3.1""). in /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php:224
Stack trace:
#0 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(454): Symfony\Component\Yaml\Parser->doParse('95', 0)
#1 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(315): Symfony\Component\Yaml\Parser->parseBlock(8, '95: "95"\n3.1: "...', 0)
#2 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(95): Symfony\Component\Yaml\Parser->doParse('windows:\n    95...', 0)
#3 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(62): Symfony\Component\Yaml\Parser->parse('###############...', 0)
#4 /home/developer/p in /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php on line 224
```

**Comment by serhiime** [comment]
@mimmi20 
What environment do you have ?
Need more details

**Comment by serhiime** [comment]
Also I need UA String for test fix this issue

**Comment by mimmi20** [comment]
@endorphinua 
> What environment do you have ?

PHP 7.1.11
Debian

**Comment by serhiime** [comment]
@mimmi20 thanx, please provide UA string for testing

**Comment by mimmi20** [comment]
The error occurs during the initialization:

```php
[2018-09-14 09:30:53] null.ERROR: Symfony\Component\Yaml\Exception\ParseException: Numeric keys are not supported. Quote your evaluable mapping keys instead at line 10 (near "3.1: "3.1""). in /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php:224
Stack trace:
#0 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(454): Symfony\Component\Yaml\Parser->doParse('95', 0)
#1 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(315): Symfony\Component\Yaml\Parser->parseBlock(8, '95: "95"\n3.1: "...', 0)
#2 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(95): Symfony\Component\Yaml\Parser->doParse('windows:\n    95...', 0)
#3 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(62): Symfony\Component\Yaml\Parser->parse('###############...', 0)
#4 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/endorphin-studio/browser-detector/src/Storage/YamlStorage.php(34): Symfony\Component\Yaml\Parser->parseFile('/home/developer...')
#5 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/endorphin-studio/browser-detector/src/Detector.php(91): EndorphinStudio\Detector\Storage\YamlStorage->getConfig()
#6 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/scripts/parse.php(35): EndorphinStudio\Detector\Detector->__construct()
#7 {main}
```

**Comment by mimmi20** [comment]
This is followed by error:

```php
[2018-09-14 10:02:23] null.ERROR: Symfony\Component\Yaml\Exception\ParseException: Duplicate key "Yandex (Direct)" detected at line 60 (near "Yandex (Direct): "(YandexDirect|YaDirectFetcher)""). in /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php:343
Stack trace:
#0 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(454): Symfony\Component\Yaml\Parser->doParse('(YandexDirect|Y...', 0)
#1 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(315): Symfony\Component\Yaml\Parser->parseBlock(48, 'Yandex (Mobile)...', 0)
#2 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(454): Symfony\Component\Yaml\Parser->doParse(Array, 0)
#3 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(315): Symfony\Component\Yaml\Parser->parseBlock(37, 'Google Inc.:\n  ...', 0)
#4 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(454): Symfony\Component\Yaml\Parser->doParse('Search Engine:\n...', 0)
#5 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(315): Symfony\Component\Yaml\Parser->parseBlock(36, 'Search Engine:\n...', 0)
#6 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(454): Symfony\Component\Yaml\Parser->doParse(Array, 0)
#7 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(315): Symfony\Component\Yaml\Parser->parseBlock(22, 'homepages:\n    ...', 0)
#8 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(95): Symfony\Component\Yaml\Parser->doParse('robot:\n    home...', 0)
#9 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/symfony/yaml/Parser.php(62): Symfony\Component\Yaml\Parser->parse('###############...', 0)
#10 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/endorphin-studio/browser-detector/src/Storage/YamlStorage.php(34): Symfony\Component\Yaml\Parser->parseFile('/home/developer...')
#11 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/vendor/endorphin-studio/browser-detector/src/Detector.php(91): EndorphinStudio\Detector\Storage\YamlStorage->getConfig()
#12 /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/scripts/parse.php(35): EndorphinStudio\Detector\Detector->__construct()
#13 {main}
```

**Comment by mimmi20** [comment]
This is followed by this error (I dont understand why this is thown):

```php
[2018-09-14 10:43:54] null.ERROR: Error: Call to undefined method EndorphinStudio\Detector\Detector::analyse() in /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/scripts/parse.php:42
Stack trace:
#0 {main}

Next Exception: an error occured while parsing ua "Test String", class EndorphinStudio\Detector\Detector, version 4 in /home/developer/projects/useragent-parser-comparison/parsers/endorphin-php-4/scripts/parse.php:44
Stack trace:
#0 {main}  
```

**Comment by mimmi20** [comment]
This is the Code:

```php
use EndorphinStudio\Detector\Detector;
use Monolog\Logger;

$logger = new Logger('null');
$formatter = new \Monolog\Formatter\LineFormatter(null, null, true, true);
$handler   = new \Monolog\Handler\StreamHandler(__DIR__ . '/../log/error.log');
$handler->setFormatter($formatter);
$logger->pushHandler($handler);
\Monolog\ErrorHandler::register($logger);

try {
    $detector = new Detector();
} catch (\Throwable $e) {
    $logger->error(new \Exception('an error occured while initializing Detector', 0, $e));
    exit;
}

try {
    $detector->analyse('Test String');
} catch (\Throwable $e) {
    $logger->error(new \Exception(sprintf('an error occured while parsing ua "%s", class %s', 'Test String', get_class($detector)), 0, $e));
    exit;
}
```

**Comment by mimmi20** [comment]
This is the composer.json:

```
{
    "require": {
        "endorphin-studio/browser-detector": "^4.0",
        "thadafinser/package-info": "^1.0",
        "monolog/monolog": "^1.23"
    }
}
```

**Comment by serhiime** [comment]
@mimmi20 I probably fix it in 4.0.1 version. Please confirm
